TestFoundation/TestPropertyListEncoder.swift

```diff
@@ -0,0 +1,130 @@
+            ("test_xmlDecoder", test_xmlDecoder),
+        ]
+    }
+}
+
+extension TestPropertyListEncoder {
+    class TestBaseClass: Codable {
+        enum IntEnum: Int, Codable, Equatable {
+            case one = 1
+            case two
+        }
+        
+        struct InnerStruct: Codable, Equatable {
+            enum StringEnum: String, Codable, Equatable {
+                case one = "1"
+                case two
+            }
+            
+            let string: String?
+            let optionalInt: Int?
+            let url: URL
+            let stringEnum: StringEnum?
+        }
+        
+        let intEnum: IntEnum
+        let innerStruct: InnerStruct
+        
+        init(intEnum: IntEnum, innerStruct: InnerStruct) {
+            self.intEnum = intEnum
+            self.innerStruct = innerStruct
+        }
+    }
+    
+    func test_basicEncodeDecode() throws {
+        let propertyListFormats: [PropertyListSerialization.PropertyListFormat?] = [nil, .binary, .xml]
+        
+        for format in propertyListFormats {
+            let optionalInt: Int? = 1234
+            let url = URL(string: "https://swift.org")!
+            let innerClassString = "demo_string"
+            
+            let innerStruct = TestBaseClass.InnerStruct(string: innerClassString, optionalInt: optionalInt, url: url, stringEnum: .two)
+            let testClass = TestBaseClass(intEnum: .one, innerStruct: innerStruct)
+            
+            let encoder = PropertyListEncoder()
+            if let format = format {
+                encoder.outputFormat = format
+            }
+            let data = try? encoder.encode(testClass)
+            XCTAssertNotNil(data)
+            
+            let decoder = PropertyListDecoder()
+            let decodedClass: TestBaseClass
+            decodedClass = try decoder.decode(TestBaseClass.self, from: data!)
+            XCTAssertEqual(decodedClass.innerStruct, testClass.innerStruct)
+            XCTAssertEqual(decodedClass.intEnum, testClass.intEnum)
+            
+            if format == .xml {
+                XCTAssertNotNil(String(data: data!, encoding: .utf8))
+            }
+        }
+    }
+}
+
+extension TestPropertyListEncoder {
+    static let propertyListXML = """
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+    <dict>
+    <key>CFBundleDevelopmentRegion</key>
+    <string>en</string>
+    <key>CFBundleExecutable</key>
+    <string>xdgTestHelper</string>
+    <key>CFBundleIdentifier</key>
+    <string>org.swift.xdgTestHelper</string>
+    <key>CFBundleInfoDictionaryVersion</key>
+    <real>6.0</real>
+    <key>CFBundleIntKey</key>
+    <integer>-100</integer>
+    <key>CFBundleBoolKey</key>
+    <true/>
+    <key>CFBundleOtherKey</key>
+    <string>other...</string>
+    </dict>
+    </plist>
```

Comment:
Can we add a test case which includes at least 1 of each property list type? `Data` and `Date` are missing here.